src/App.jsx

```diff
@@ -14,45 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+// let preparedGoodsList = [...goodsFromServer];
+
+const SORT_TYPE = { length: 'length', alphabet: 'alphabet' };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
+export const GoodsList = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li data-cy="Good" key={good}>{good}</li>
+    ))}
+  </ul>
 );
+
+function sortPreparedGoods(arr, sortBy, reverse) {
+  let preparedGoodsList = [...arr];
+
+  switch (sortBy) {
+    case SORT_TYPE.length:
+      preparedGoodsList.sort((a, b) => a.length - b.length);
+      break;
+    case SORT_TYPE.alphabet:
+      preparedGoodsList.sort((a, b) => a.localeCompare(b));
+      break;
+    default: preparedGoodsList = [...goodsFromServer];
+  }
+
+  if (reverse === 'reverse') {
```

Comment:
you can write 
`if (reverse) {`

if `reverse` would be `empty string, 0 or null` it will be false, otherwise - true

also I think it would be better to use boolean value for `reverse` instead of string and call this value `isReverse`